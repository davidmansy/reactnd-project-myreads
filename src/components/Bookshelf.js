import React from 'react';
import BooksGrid from './BooksGrid';

function Bookshelf(props) {
  const { shelf, books, handleShelfChange, loading } = props;
  const shelfBooks = books.filter(book => {
    return book.shelf === shelf.id;
  });

  return (
    <div className="bookshelf">
      <h2 className="bookshelf-title">{shelf.title}</h2>
      <div className="bookshelf-books">
        {loading ? (
          <div>Loading...</div>
        ) : shelfBooks.length > 0 ? (
          <BooksGrid books={shelfBooks} handleShelfChange={handleShelfChange} />
        ) : (
          <div>There are not books available in this shelf</div>
        )}
      </div>
    </div>
  );
}

export default Bookshelf;
