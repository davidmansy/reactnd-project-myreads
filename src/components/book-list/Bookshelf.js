import React from 'react';
import BooksGrid from '../books/BooksGrid';

function Bookshelf(props) {
  const { shelf, books, handleShelfChange } = props;
  const shelfBooks = books.filter(book => {
    return book.shelf === shelf.id;
  });

  return (
    <div className="bookshelf">
      <h3 className="bookshelf-title">{shelf.title}</h3>
      <div className="bookshelf-books">
        {shelfBooks.length > 0 ? (
          <BooksGrid books={shelfBooks} handleShelfChange={handleShelfChange} />
        ) : (
          <div className="bookshelf-empty">
            There are not books available in this shelf
          </div>
        )}
      </div>
    </div>
  );
}

export default Bookshelf;
