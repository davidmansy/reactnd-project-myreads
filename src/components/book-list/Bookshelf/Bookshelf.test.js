import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Bookshelf from './Bookshelf';
import { MemoryRouter } from 'react-router-dom';
import { cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('Bookshelf', () => {
  const props = {
    shelf: {},
    books: [],
    handleShelfChange: () => {}
  };

  it('renders', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Bookshelf {...props} />
      </MemoryRouter>,
      container
    );
  });

  it('snapshots', () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <Bookshelf {...props} />
      </MemoryRouter>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
