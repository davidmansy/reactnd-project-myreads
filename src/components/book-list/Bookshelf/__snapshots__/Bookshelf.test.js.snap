// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookshelf snapshots 1`] = `
<div
  className="bookshelf"
>
  <h3
    className="bookshelf-title"
  />
  <div
    className="bookshelf-books"
  >
    <div
      className="bookshelf-empty"
    >
      There are not books available in this shelf
    </div>
  </div>
</div>
`;
